File: tests/fixtures/typescript_api_app/src/services/userService.ts
export function databaseUrl(): string {
  return process.env.DATABASE_URL ?? "sqlite://memory";
}

export function createUser(name: string): string {
  databaseUrl();
  return name;
}

// Never called: pins that an unreferenced export gets no inbound CALLS edge.
export function deleteUser(name: string): boolean {
  return name.length > 0;
}

File: tests/fixtures/java_service_app/src/main/java/com/example/services/UserService.java
package com.example.services;

public class UserService {

    public static String createUser(String name) {
        return name;
    }

    // Never referenced anywhere: pins that an uncalled method gets no
    // inbound CALLS edge.
    public static boolean deleteUser(String name) {
        return name != null;
    }

    public String describe() {
        return "user service";
    }
}
